enhancing/enhancer.js

```diff
@@ -6,14 +6,26 @@ module.exports = {
 };
 
 function succeed(item) {
+  if (item.enhancement < 20) {
+    return { ...item, enhancement: item.enhancement + 1 }
+  } else {
   return { ...item };
+  }
 }
 
 function fail(item) {
-  return { ...item };
+  if (item.enhancement >= 15) {
+    if (item.enhancement > 16) {
+      return { ...item, enhancement: item.enhancement -1 }
+    }
+    return { ...item, durability: item.durability - 10 }
```

Comment:
tried using { ...item, enhancement: Math.max(1, item.enhancement -1)} so you couldn't go below 1 durability, but didn't seem to work in testing